drivers/isl29125/isl29125.c

```diff
@@ -0,0 +1,272 @@
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     driver_isl29125
+ * @{
+ *
+ * @file
+ * @brief       Device driver implementation for the ISL29125 RGB light sensor
+ *
+ * @author      Ludwig Ortmann <ludwig.ortmann@fu-berlin.de>
+ *
+ * @}
+ */
+
+#include <stdint.h>
+
+#include "isl29125.h"
+#include "isl29125-internal.h"
+#include "periph/i2c.h"
+
+#define ENABLE_DEBUG    (0)
+#include "debug.h"
+
+/***********************************************************************
+ * internal API declaration
+ **********************************************************************/
+
+static inline char parse_mode(isl29125_mode_t mode);
+static inline char parse_range(isl29125_range_t range);
+static inline char parse_resolution(isl29125_resolution_t resolution);
+static inline char parse_conf1(isl29125_mode_t mode, isl29125_range_t range,
+        isl29125_resolution_t resolution);
+
+/***********************************************************************
+ * internal API implementation
+ **********************************************************************/
+
+static inline char parse_mode(isl29125_mode_t mode)
+{
+    char ret = 0x00;
+
+    switch (mode) {
+        case ISL29125_MODE_DOWN:
+            ret |= ISL29125_CON1_POWERDOWN;
+            break;
+
+        case ISL29125_MODE_STANDBY:
+            ret |= ISL29125_CON1_STANDBY;
+            break;
+
+        case ISL29125_MODE_RGB:
+            ret |= ISL29125_CON1_GRB;
+            break;
+
+        case ISL29125_MODE_R:
+            ret |= ISL29125_CON1_R;
+            break;
+
+        case ISL29125_MODE_G:
+            ret |= ISL29125_CON1_G;
+            break;
+
+        case ISL29125_MODE_B:
+            ret |= ISL29125_CON1_B;
+            break;
+
+        case ISL29125_MODE_RG:
+            ret |= ISL29125_CON1_GR;
+            break;
+
+        case ISL29125_MODE_GB:
+            ret |= ISL29125_CON1_GB;
+            break;
+
+        default:
+            DEBUG("parse_mode: invalid mode\n");
+            break;
+    }
+
+    return ret;
+}
+
+static inline char parse_range(isl29125_range_t range)
+{
+    char ret = 0x00;
+
+    switch (range) {
+        case ISL29125_RANGE_375:
+            ret |= ISL29125_CON1_375LUX;
+            break;
+
+        case ISL29125_RANGE_10K:
+            ret |= ISL29125_CON1_10KLUX;
+            break;
+
+        default:
+            DEBUG("parse_range: invalid range\n");
+            break;
+    }
+
+    return ret;
+}
+
+static inline char parse_resolution(isl29125_resolution_t resolution)
+{
+    char ret = 0x00;
+
+    switch (resolution) {
+        case ISL29125_RESOLUTION_12:
+            ret |= ISL29125_CON1_12BRES;
+            break;
+
+        case ISL29125_RESOLUTION_16:
+            ret |= ISL29125_CON1_16BRES;
+            break;
+
+        default:
+            DEBUG("parse_resolution: invalid resolution\n");
+            return -1;
```

Comment:
Ignore when going for @haukepetersen's proposal above (which may lead to a smaller implementation). Why didn't you use default values in the other static functions?